### PR TITLE
master

### PR DESCRIPTION
**Pull request by TheJhonny007**
- Property.plus(), minus(), etc... now return Bindings instead of Properties

- Added Extensions for StringProperty and BooleanPropert

### CODE REVIEW

**Comment by edvin** [comment]
This sounds like a good idea, and even though we break the API I think it should be done. It will take me a little while to go through the PR though, will get back to you ASAP.

**Comment by TheJhonny007** [comment]
Great, I was defenitly too fast with the first version, because it was a terrible idea to return Properties instead of Bindings

**Comment by edvin** [comment]
Hehe, yeah sorry I didn't catch that :)

**Comment by edvin** [comment]
It seems like you have double the indentation in the files, so the diff is pretty big. Can you revert the spacing changes so the diff only contains code changes?

**Comment by TheJhonny007** [comment]
Sorry didn't notice you use spaces for indentation and IntelliJ automatically replaced them with tabs when I auto formatted my code

File: src/main/java/tornadofx/Properties.kt [hunk 1]
```diff
@@ -225,89 +225,50 @@ operator fun IntegerProperty.setValue(thisRef: Any, property: KProperty<*>, valu
...
-        = set(get() + other.doubleValue())
+operator fun DoubleExpression.plus(other: Number): DoubleBinding = add(other.toDouble())
+operator fun DoubleExpression.plus(other: ObservableNumberValue): DoubleBinding = add(other)
 
+operator fun DoubleProperty.plusAssign(other: Number) = set(get() + other.toDouble())
+operator fun DoubleProperty.plusAssign(other: ObservableNumberValue) = set(get() + other.doubleValue())
 
 operator fun DoubleProperty.inc(): DoubleProperty {
-    set(get() + 1.0)
+    set(get() + 1)
```
Comment (t-boom) [inline src/main/java/tornadofx/Properties.kt]: This might be clearer as `value += 1.0`, but that may just be me.

**Comment by t-boom** [comment]
There are a number of places where you have code like `set(get() + inc)`. Personally I think it would be clearer to use `value += inc`. but I'm curious to know if I'm the only one.

(This also goes for other operators, e.g. `set(get() - dec)` => `value -= dec`, `set(get() * mul)` => `value *= mul`, etc.)

**Comment by edvin** [comment]
Yeah, it even seems `value` would be slightly more correct if I understand the semantics correctly.

**Comment by TheJhonny007** [comment]
That is only possible if you convert the functions to functions with bodies, since only expressions and no assignments are allowed inline. Otherwise you get this error:

`Error:(r, c) Kotlin: Assignments are not expressions, and only expressions are allowed in this context`

I could expand all those functions, but that would bloat the API.

**Comment by edvin** [comment]
Thank you for this :) Merging now.

**Pull request merged**